source/h2_connection.c

```diff
@@ -310,6 +313,31 @@ static struct aws_h2_connection *s_connection_new(
         goto error;
     }
 
+    /* Initial settings with initial window size and disable push promise for the client */
+    struct aws_h2_frame_setting initial_settings[2];
+    size_t iter = 0;
+    if (!server) {
+        initial_settings[iter].id = AWS_H2_SETTINGS_ENABLE_PUSH;
+        initial_settings[iter++].value = 0;
+    }
+    if (initial_window_size != aws_h2_settings_initial[AWS_H2_SETTINGS_INITIAL_WINDOW_SIZE]) {
+        initial_settings[iter].id = AWS_H2_SETTINGS_INITIAL_WINDOW_SIZE;
+        initial_settings[iter++].value = initial_window_size;
```

Comment:
We will not correct this initial_window_size for user, if the user gives us a number that is too big. Since I think user will need that value for manually flow control. But, we can probably assert/precondition to make sure this value will be among the required bounds.